### ISSUE TITLE
GetPurchaseAsync only returns first purchase of a subscription.

### ISSUE THREAD

**Issue by kharhui** [jamesmontemagno/InAppBillingPlugin#321]
I am having issue on getting purchase history of a subscription in **Android** only. I have tested on iOS and the code works as expected.

## Bug Information

Version Number of Plugin:2.0
Device Tested On:-
Simulator Tested On: OnePlus A5000
Version of VS: 2019 v16.7.4
Version of Xamarin: 4.5.0.617
Versions of other things you are using:

### Steps to reproduce the Behavior
In a Android testing environment, configure subscription expires every 5 minute. 
In the app, make a purchase of a subscription and make sure auto-renewal has been enabled. 
Now wait for 2nd purchase to be made after 5 minute.
After 5 minutes of waiting, there are 2 items in the purchase history on my Google Play account. Now call GetPurchaseAsync() , it returns the initial purchase only, which is an IEnumerable with count = 1. 
I have checked with the order id, the order id does not end with ..0 or ..1 even after a few auto-renew of my subscription has been made. 

### Expected Behavior
GetPurchaseAsync should returns all purchases of my subscription, instead of the initial purchase only.

### Actual Behavior
GetPurchaseAsync returns single purchase and it is the initial purchase.

**Comment by InquisitorJax** [comment]
As far as I understand it - on Android - you will only get ACTIVE purchases returned from the call.
So if your purchase is not in the list, you can assume is has expired.
With iOS - you get the whole history of purchases - but no way to know which one of those purchases has expired: You need to validate the receipt to check that.

**Comment by SerkanKrkc** [comment]
@InquisitorJax

This is how we verify for Android and successfully completed all our tests.

purchase = await billing.PurchaseAsync(dergi.Id.ToString(), ItemType.InAppPurchase);
                        if (purchase != null && purchase.State == PurchaseState.Purchased)
                        {
                           // var cc = await billing.ConsumePurchaseAsync(dergi.Id.ToString(), purchase.PurchaseToken);
                            var ack = await billing.AcknowledgePurchaseAsync(purchase.PurchaseToken);
                            if (ack)
                            {
                               await DergiIndir(dergi);
                            }
                            
                        }

How exactly do we need to verify for iOS?

**Comment by InquisitorJax** [comment]
@SerkanKrkc check out this thread: https://github.com/jamesmontemagno/InAppBillingPlugin/issues/311
basically - iOS will give you a list of purchases - you then loop through those and return only the ones that have an expires-date in the future.

**Comment by SerkanKrkc** [comment]
@InquisitorJax 

Thanks for the answer, I have studied the relevant topic and even I have a question there, not yet answered.

Actually, this is exactly my question; Play store returns a purchased product or subscription if it is not approved within 3 days. Does Apple do something like this? Does this require an extra confirmation?

On the other hand, can you help me with my unanswered question there?

```
Do you have any idea how the results below work in a production environment? Have you had a chance to test it?

I am particularly curious about the AutoRenewing and IsAcknowledged result.

Thanks in advance.

"Id": "xxxxxxxxxxxxxxxx",
"TransactionDateUtc": "2021-03-29T20:21:54Z",
"ProductId": "xxx",
"AutoRenewing": false,
"PurchaseToken": "xxxxxxxxxxxxxxxxxxx",
"State": 0,
"ConsumptionState": 0,
"IsAcknowledged": false,
"Payload": null
```

**Comment by Erdogan34** [comment]
@SerkanKrkc did you find solution for ios?

**Comment by SerkanKrkc** [comment]
@Erdogan34 

Yes, I created a solution. I'm looking to see if the user has made a purchase or subscription. I pull the list with GetPurchasesAsync and compare it with my own product model. I check "AutoRenewStatus == 1" if subscribed for the product I found. 

If not, I check "response.LatestReceiptInfo.OriginalPurchaseDateUtc == response.Receipt.PurchaseDateUtc" and check if it's still the authorized person.

So there is no standard for the job. We adapted it to our own scenario.

**Comment by Erdogan34** [comment]
@SerkanKrkc , thank you for answering.

Thank you for answering. Then AutoRenewStatus doesn't come false always in production, because it's always return false in sandbox.

Well, If an user cancel the subscription, will IsAcknowledged return false? How can i know an user still active or not?

**Comment by SerkanKrkc** [comment]
Yes, AutoRenewStatus returns according to the situation in production.

If the user has closed the renewal or canceled the subscription; We look at the subscription expiry date and act accordingly.

**Comment by Erdogan34** [comment]
@SerkanKrkc , thanks again. Everything is clear now.

**Comment by SerkanKrkc** [comment]
> @SerkanKrkc , thanks again. Everything is clear now.

Good luck. :)

**Comment by jamesmontemagno** [comment]
Here is a long sample of what I am currently testing out for both normal purchases and subscriptions. Things are working well for me.

https://gist.github.com/jamesmontemagno/e9502b7605f4068f644d3cfc5fe0f19f

**Comment by Erdogan34** [comment]
Great lines for me thanks James..
